signer/src/message.rs

```diff
@@ -41,3 +46,67 @@ pub struct BitcoinTransactionSignRequest;
 
 #[derive(Debug, Clone, PartialEq)]
 pub struct BitcoinTransactionSignAck;
+
+impl wsts::net::Signable for SignerMessage {
+    fn hash(&self, hasher: &mut sha2::Sha256) {
+        hasher.update("SBTC_SIGNER_MESSAGE");
+        hasher.update(self.bitcoin_chain_tip);
+        self.payload.hash(hasher);
+    }
+}
+
+impl wsts::net::Signable for Payload {
+    fn hash(&self, hasher: &mut sha2::Sha256) {
+        match self {
+            Self::WstsMessage(msg) => hash_message(msg, hasher),
+            _ => unimplemented!(),
+        }
+    }
+}
+
+/// Utility method because wsts::net::Message doesn't implement wsts::net::Signable
+fn hash_message(msg: &wsts::net::Message, hasher: &mut sha2::Sha256) {
+    use wsts::net::Signable;
+
+    match msg {
+        wsts::net::Message::DkgBegin(msg) => msg.hash(hasher),
+        wsts::net::Message::DkgPublicShares(msg) => msg.hash(hasher),
+        wsts::net::Message::DkgPrivateBegin(msg) => msg.hash(hasher),
+        wsts::net::Message::DkgPrivateShares(msg) => msg.hash(hasher),
+        wsts::net::Message::DkgEndBegin(msg) => msg.hash(hasher),
+        wsts::net::Message::DkgEnd(msg) => msg.hash(hasher),
+        wsts::net::Message::NonceRequest(msg) => msg.hash(hasher),
+        wsts::net::Message::NonceResponse(msg) => msg.hash(hasher),
+        wsts::net::Message::SignatureShareRequest(msg) => msg.hash(hasher),
+        wsts::net::Message::SignatureShareResponse(msg) => msg.hash(hasher),
+    }
```

Comment:
Is this because the `hash` function is different in all of them from the compiler's perspective?

Reply:
Because the inner class is different for each of them

Reply:
This is to dispatch the `.hash` call to all variants. In the next version `wsts` will implement `Signable` directly on the `Message` type so we don't have to do this here.

I know this looks quite repetitive, and you can get around this type of code by using for example the `enum_dispatch` crate: https://crates.io/crates/enum_dispatch.